src/app/components/course-island/course-island.component.html

```diff
@@ -18,17 +18,27 @@ <h2 class="tui-island__title">{{ course.name }}</h2>
         ></tui-tag>
         <div class="tui-form__buttons">
             <a
-                *ngIf="hasViewPermission"
+                *ngIf = "endDate < currentDate && user.is_teacher"
+                appearance="primary"
                 [routerLink]="['/course', course.id]"
+                class="tui-form__button"
+                size="m"
+                tuiButton
+            >
+                View
+            </a>
+            <a
+                *ngIf="hasViewPermission && endDate > currentDate"
                 appearance="primary"
+                [routerLink]="['/course', course.id]"
                 class="tui-form__button"
                 size="m"
                 tuiButton
             >
                 Open
             </a>
             <a
-                *ngIf="!course.is_registered"
+                *ngIf="!course.is_registered && endDate > currentDate"
```

Comment:
Why are we showing `Register` when the course in finished? This is not the correct behaviour.

Reply:
endDate > currentDate shows that the course is ongoing

Reply:
Yeah sorry I misread that =))